Code/GraphMol/Descriptors/AUTOCORR3D.cpp

```diff
@@ -0,0 +1,242 @@
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[0][i]=dtmp;
+
+    tmp=Wm.transpose() * RBi * Wm / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[1][i]=dtmp;
+
+    tmp=Wv.transpose() * RBi * Wv / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[2][i]=dtmp;
+
+    tmp=We.transpose() * RBi * We / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[3][i]=dtmp;
+
+    tmp=Wp.transpose() * RBi * Wp / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[4][i]=dtmp;
+
+    tmp=Wi.transpose() * RBi * Wi / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[5][i]=dtmp;
+
+    tmp=Ws.transpose() * RBi * Ws / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[6][i]=dtmp;
+
+    tmp=Wr.transpose() * RBi * Wr / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[7][i]=dtmp;
+
+  }
+
+  // create the Output vector!
+    w= AppendDouble(w, TDBmat[0], 10, 0);
+    w= AppendDouble(w, TDBmat[1], 10, 10);
+    w= AppendDouble(w, TDBmat[2], 10, 20);
+    w= AppendDouble(w, TDBmat[3], 10, 30);
+    w= AppendDouble(w, TDBmat[4], 10, 40);
+    w= AppendDouble(w, TDBmat[5], 10, 50);
+    w= AppendDouble(w, TDBmat[6], 10, 60);
+    w= AppendDouble(w, TDBmat[7], 10, 70);
+
+    return w;
+
+}
+
+  double* Get3Dauto(double* dist3D,double* dist, int numAtoms, const ROMol& mol){
+
+      std::vector<std::string> AUTOCORRNAMES={"TDB01u","TDB02u","TDB03u","TDB04u","TDB05u","TDB06u","TDB07u","TDB08u","TDB09u","TDB10u","TDB01m","TDB02m","TDB03m","TDB04m","TDB05m","TDB06m","TDB07m","TDB08m","TDB09m","TDB10m","TDB01v","TDB02v","TDB03v","TDB04v","TDB05v","TDB06v","TDB07v","TDB08v","TDB09v","TDB10v","TDB01e","TDB02e","TDB03e","TDB04e","TDB05e","TDB06e","TDB07e","TDB08e","TDB09e","TDB10e","TDB01p","TDB02p","TDB03p","TDB04p","TDB05p","TDB06p","TDB07p","TDB08p","TDB09p","TDB10p","TDB01i","TDB02i","TDB03i","TDB04i","TDB05i","TDB06i","TDB07i","TDB08i","TDB09i","TDB10i","TDB01s","TDB02s","TDB03s","TDB04s","TDB05s","TDB06s","TDB07s","TDB08s","TDB09s","TDB10s","TDB01r","TDB02r","TDB03r","TDB04r","TDB05r","TDB06r","TDB07r","TDB08r","TDB09r","TDB10r"};
+      double *res= new double[80];
+      res= get3DautocorrelationDesc(dist3D,dist, numAtoms, mol);
+      return res;
+  }
+
+} //end of anonymous namespace
+
+
+std::vector<double> AUTOCORR3D(const ROMol& mol,int confId){
+  PRECONDITION(mol.getNumConformers()>=1,"molecule has no conformers")
+  int numAtoms = mol.getNumAtoms();
+
+  const Conformer &conf = mol.getConformer(confId);
+
+  double *dist = MolOps::getDistanceMat(mol, false);
+  double *dist3D = MolOps::get3DDistanceMat(mol, confId);
+  double *wpol= Get3Dauto(dist3D, dist, numAtoms, mol);
+
+  std::vector<double> dataVec;
+  for (int i=0;i<80;i++){
+    dataVec.push_back(wpol[i]);
+  }
```

Comment:
@gedeck I was looking at an expanded diff, and saw:

```
   double *dist = MolOps::getDistanceMat(mol, false);
   double *dist3D = MolOps::get3DDistanceMat(mol, confId);
```

These aren't deleted and neither is Get3Dauto.

For your own code, I *highly* suggest using std::vector<double> as opposed to pointers.  They have no overhead to return from a function in C++03 (copy elision)  or C++11 (std::move) as follows:


```
// automatically deleted when out of scope
std::vector<float> v = Get3Dauto(...);

```

They are only used in the RDKit API points for historical reasons and can cause a lot of issues.  In the future I expect them to be converted to std::vector<double> or unique_pointer<double> for automatic memory cleanup.